### ISSUE TITLE
Index out of bounds while editing build.zig.zon

### ISSUE THREAD

**Issue by pfgithub** [zigtools/zls#2120]
### Zig Version

0.14.0-dev.2563+af5e73172

### ZLS Version

0.14.0-dev.308+34e0fd5

### Client / Code Editor / Extensions

vscode Zig Language

### Steps to Reproduce and Observed Behavior

Edit build.zig.zon, paste this in:

```
.{
    .a = "b
}
```

### Expected Behavior

Shouldn't crash

### Relevant log output

```shell
Took 49ms to process request-0-initialize on Thread 10957659
Dynamically registering method 'workspace/didChangeConfiguration'
Took 0ms to process notification-initialized on Thread 10957659
[Info  - 10:47:07 PM] Loaded build file 'file:///Users/pfg/Dev/Node/blockeditor/packages/tracy/build.zig'
[Info  - 10:47:07 PM] Loaded build file 'file:///Users/pfg/Dev/Node/blockeditor/build.zig'
Opened document 'file:///Users/pfg/Dev/Node/blockeditor/packages/tracy/build.zig.zon'
Took 0ms to process notification-textDocument/didOpen on Thread 10957659
Took 0ms to process request-1-textDocument/completion on Thread 10957659
Took 0ms to process response-"register-workspace/didChangeConfiguration" on Thread 10957659
[Info  - 10:47:07 PM] Set config option 'semantic_tokens' to "partial"
[Info  - 10:47:07 PM] trying to start Build-On-Save for 'file:///Users/pfg/Dev/Node/blockeditor'
thread 10957663 panic: index out of bounds: index 2863311530, len 12
Took 19ms to process response-"i_haz_configuration" on Thread 10957659
Took 0ms to process request-3-textDocument/codeAction on Thread 10957662
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/zig/Ast.zig:483:30: 0x10009c5af in firstToken (zls)
    while (true) switch (tags[n]) {
                             ^
/Users/pfg/Dev/Node/zls/src/offsets.zig:500:45: 0x1000f479b in nodeToLoc (zls)
    return tokensToLoc(tree, tree.firstToken(node), ast.lastToken(tree, node));
                                            ^
/Users/pfg/Dev/Node/zls/src/ast.zig:1928:67: 0x100159b5b in inner (zls)
            try self.locs.append(self.allocator, offsets.nodeToLoc(ast, node));
                                                                  ^
/Users/pfg/Dev/Node/zls/src/ast.zig:1577:29: 0x1000de89b in nodesAtLoc (zls)
                try callback(context, tree, child);
                            ^
/Users/pfg/Dev/Node/zls/src/features/inlay_hints.zig:592:37: 0x100063673 in sendRequestSync__anon_20828 (zls)
    const nodes = try ast.nodesAtLoc(arena, handle.tree, loc);
                                    ^
/Users/pfg/Dev/Node/zls/src/Server.zig:2074:58: 0x100041c1f in processMessage (zls)
                const result = try server.sendRequestSync(arena_allocator.allocator(), @tagName(method), params);
                                                         ^
/Users/pfg/Dev/Node/zls/src/Server.zig:2088:33: 0x10001e1ff in processJob (zls)
    return server.processMessage(message) catch |err| {
                                ^
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/Thread/Pool.zig:115:39: 0x100031d27 in runFn (zls)
            @call(.auto, func, closure.arguments);
                                      ^
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/Thread/Pool.zig:291:32: 0x10002f72b in entryFn (zls)
            run_node.data.runFn(&run_node.data, id);
                               ^
???:?:?: 0x195e9af93 in ??? (libsystem_pthread.dylib)
???:?:?: 0xab1b000195e95d33 in ??? (???)
thread 10957662 panic: index out of bounds: index 2863311530, len 12
/Users/pfg/Dev/Node/zls/src/features/document_symbol.zig:38:52: 0x100127b4f in inner (zls)
    const maybe_symbol: ?Symbol = switch (node_tags[node]) {
                                                   ^
/Users/pfg/Dev/Node/zls/src/ast.zig:1577:29: 0x1000f17f7 in iterateChildren__anon_56110 (zls)
                try callback(context, tree, child);
                            ^
/Users/pfg/Dev/Node/zls/src/features/document_symbol.zig:246:28: 0x10006951f in sendRequestSync__anon_21354 (zls)
    try ast.iterateChildren(tree, 0, &ctx, error{OutOfMemory}, callback);
                           ^
/Users/pfg/Dev/Node/zls/src/Server.zig:2074:58: 0x100041def in processMessage (zls)
                const result = try server.sendRequestSync(arena_allocator.allocator(), @tagName(method), params);
                                                         ^
/Users/pfg/Dev/Node/zls/src/Server.zig:2088:33: 0x10001e1ff in processJob (zls)
    return server.processMessage(message) catch |err| {
                                ^
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/Thread/Pool.zig:115:39: 0x100031d27 in runFn (zls)
            @call(.auto, func, closure.arguments);
                                      ^
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/Thread/Pool.zig:291:32: 0x10002f72b in entryFn (zls)
            run_node.data.runFn(&run_node.data, id);
                               ^
???:?:?: 0x195e9af93 in ??? (libsystem_pthread.dylib)
???:?:?: 0xbe68000195e95d33 in ??? (???)
[Error - 10:47:07 PM] The ZLS language server server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.


```
```

**Comment by Techatrix** [comment]
Does #2121 resolve your issue? I haven't been able to reproduce the exact same crash but did find some similar ones.

**Comment by pfgithub** [comment]
Still crashes unfortunately. Interestingly it seems to be only crashing in ReleaseSafe? I can't reproduce it in Debug or ReleaseFast. I aso tried clearing `.zig-cache` but it still crashes with a ReleaseSafe build.

I can't reproduce it with a debug build or a releasefast build

<details>
<summary>More reproduction information</summary>

Build zls:

```
git checkout techatrix/fix-zon-mode
zig build -Doptimize=ReleaseSafe
```

Edit made to an open file:

```
.{
    .a = "b",
}
```

press backspace twice from `",`. it crashes when the quote is deleted:

```
Took 0ms to process notification-textDocument/didChange on Thread 11073812
Took 0ms to process request-6-textDocument/inlayHint on Thread 11073815
Took 0ms to process request-7-textDocument/foldingRange on Thread 11073813
Took 0ms to process request-8-textDocument/codeAction on Thread 11073815
Took 0ms to process request-9-textDocument/documentSymbol on Thread 11073813
Took 0ms to process request-10-textDocument/documentSymbol on Thread 11073815
Took 0ms to process request-11-textDocument/inlayHint on Thread 11073813
Took 0ms to process notification-textDocument/didChange on Thread 11073812
thread 11073813 panic: index out of bounds: index 2863311530, len 1
/Users/pfg/Dev/Node/zls/src/features/inlay_hints.zig:430:26: 0x1043c1523 in writeNodeInlayHint (zls)
    const tag = node_tags[node];
                         ^
/Users/pfg/Dev/Node/zls/src/features/inlay_hints.zig:595:31: 0x1043436a7 in sendRequestSync__anon_20828 (zls)
        try writeNodeInlayHint(&builder, handle.tree, child);
                              ^
/Users/pfg/Dev/Node/zls/src/Server.zig:2075:58: 0x104321c1f in processMessage (zls)
                const result = try server.sendRequestSync(arena_allocator.allocator(), @tagName(method), params);
                                                         ^
/Users/pfg/Dev/Node/zls/src/Server.zig:2089:33: 0x1042fe1ff in processJob (zls)
    return server.processMessage(message) catch |err| {
                                ^
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/Thread/Pool.zig:115:39: 0x104311d27 in runFn (zls)
            @call(.auto, func, closure.arguments);
                                      ^
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/Thread/Pool.zig:291:32: 0x10430f72b in entryFn (zls)
            run_node.data.runFn(&run_node.data, id);
                               ^
???:?:?: 0x195e9af93 in ??? (libsystem_pthread.dylib)
???:?:?: 0x8e7c800195e95d33 in ??? (???)
[Info  - 12:27:07 AM] Connection to server got closed. Server will restart.
```

Executing Start/Restart ZLS with the file open

```
.{
    .a = "b
}

```

```
[Info  - 12:28:31 AM] Starting ZLS      0.14.0-dev.309+3c3c6d8 @ '/Users/pfg/Apps/bin/zls'
[Info  - 12:28:31 AM] Log File:         /Users/pfg/Library/Caches/zls/zls.log (info)
[Info  - 12:28:31 AM] Client Info:      Visual Studio Code-1.96.2
[Info  - 12:28:31 AM] Autofix Mode:     source.fixall
Offset Encoding:  utf-16
[Info  - 12:28:31 AM] added Workspace Folder: file:///Users/pfg/Dev/Node/blockeditor
[Info  - 12:28:31 AM] Set config option 'builtin_path' to "/Users/pfg/Library/Caches/zls/builtin.zig"
[Info  - 12:28:31 AM] Set config option 'zig_lib_path' to "/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib"
[Info  - 12:28:31 AM] Set config option 'zig_exe_path' to "/Users/pfg/Apps/bin/zig"
[Info  - 12:28:31 AM] Set config option 'build_runner_path' to "/Users/pfg/Library/Caches/zls/build_runner/31755aaa5daba3be3b3e2b608dc8531e/build_runner.zig"
[Info  - 12:28:31 AM] Set config option 'global_cache_path' to "/Users/pfg/Library/Caches/zls"
Took 50ms to process request-0-initialize on Thread 11074988
Dynamically registering method 'workspace/didChangeConfiguration'
Took 0ms to process notification-initialized on Thread 11074988
[Info  - 12:28:31 AM] Loaded build file 'file:///Users/pfg/Dev/Node/blockeditor/packages/blocks/build.zig'
Opened document 'file:///Users/pfg/Dev/Node/blockeditor/packages/blocks/build.zig' (build file)
Took 0ms to process notification-textDocument/didOpen on Thread 11074988
[Info  - 12:28:31 AM] Loaded build file 'file:///Users/pfg/Dev/Node/blockeditor/packages/tracy/build.zig'
[Error - 12:28:31 AM] Failed to execute build runner to collect build configuration, command:
/Users/pfg/Apps/bin/zig build --build-runner /Users/pfg/Library/Caches/zls/build_runner/31755aaa5daba3be3b3e2b608dc8531e/build_runner.zig
Error: /Users/pfg/Dev/Node/blockeditor/packages/blocks/../tracy/build.zig.zon:2:10: error: expected expression, found 'invalid token'

[Info  - 12:28:31 AM] Loaded build file 'file:///Users/pfg/Dev/Node/blockeditor/build.zig'
[Error - 12:28:31 AM] Failed to load build configuration for file:///Users/pfg/Dev/Node/blockeditor/packages/blocks/build.zig (error: error.RunFailed)
Opened document 'file:///Users/pfg/Dev/Node/blockeditor/packages/tracy/build.zig.zon'
Took 0ms to process notification-textDocument/didOpen on Thread 11074988
Took 0ms to process response-"register-workspace/didChangeConfiguration" on Thread 11074988
[Info  - 12:28:31 AM] Set config option 'semantic_tokens' to "partial"
[Error - 12:28:31 AM] Failed to execute build runner to collect build configuration, command:
/Users/pfg/Apps/bin/zig build --build-runner /Users/pfg/Library/Caches/zls/build_runner/31755aaa5daba3be3b3e2b608dc8531e/build_runner.zig
Error: /Users/pfg/Dev/Node/blockeditor/packages/tracy/build.zig.zon:2:10: error: expected expression, found 'invalid token'

thread 11074989 panic: index out of bounds: index 2863311530, len 1
[Error - 12:28:31 AM] Failed to load build configuration for file:///Users/pfg/Dev/Node/blockeditor/packages/tracy/build.zig (error: error.RunFailed)
[Info  - 12:28:31 AM] trying to start Build-On-Save for 'file:///Users/pfg/Dev/Node/blockeditor'
Took 22ms to process response-"i_haz_configuration" on Thread 11074988
Took 0ms to process request-1-textDocument/documentSymbol on Thread 11074992
[Error - 12:28:31 AM] Failed to execute build runner to collect build configuration, command:
/Users/pfg/Apps/bin/zig build --build-runner /Users/pfg/Library/Caches/zls/build_runner/31755aaa5daba3be3b3e2b608dc8531e/build_runner.zig
Error: /Users/pfg/Dev/Node/blockeditor/packages/tracy/build.zig.zon:2:10: error: expected expression, found 'invalid token'

zig build runner process has exited
/Users/pfg/Dev/Node/zls/src/features/inlay_hints.zig:430:26: 0x100159523 in writeNodeInlayHint (zls)
    const tag = node_tags[node];
                         ^
/Users/pfg/Dev/Node/zls/src/features/inlay_hints.zig:595:31: 0x1000db6a7 in sendRequestSync__anon_20828 (zls)
        try writeNodeInlayHint(&builder, handle.tree, child);
                              ^
/Users/pfg/Dev/Node/zls/src/Server.zig:2075:58: 0x1000b9c1f in processMessage (zls)
                const result = try server.sendRequestSync(arena_allocator.allocator(), @tagName(method), params);
                                                         ^
/Users/pfg/Dev/Node/zls/src/Server.zig:2089:33: 0x1000961ff in processJob (zls)
    return server.processMessage(message) catch |err| {
                                ^
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/Thread/Pool.zig:115:39: 0x1000a9d27 in runFn (zls)
            @call(.auto, func, closure.arguments);
                                      ^
/Users/pfg/Apps/zig/zig-macos-aarch64-0.14.0-dev.2563+af5e73172/lib/std/Thread/Pool.zig:291:32: 0x1000a772b in entryFn (zls)
            run_node.data.runFn(&run_node.data, id);
                               ^
???:?:?: 0x195e9af93 in ??? (libsystem_pthread.dylib)
???:?:?: 0xe576000195e95d33 in ??? (???)
[Info  - 12:28:31 AM] Connection to server got closed. Server will restart.
```

</details>

**Comment by Techatrix** [comment]
I managed to reproduce the issue with a ReleaseSafe build. I disabled most of affected requests zon files in #2121 (they were useless anyway). Fixing semantic tokens will require a change in the parser. Until that happens, it has also been disabled on syntactically invalid files.

**Comment by Techatrix** [comment]
This should be fixed by #2121.
